src/harness/testcases/WINNF_FT_S_SCS_testcase.py

```diff
@@ -256,136 +257,137 @@ def test_WINNF_FT_S_SCS_15(self,config_filename):
+    self.assertTlsHandshakeFailure(client_cert=device_cert, client_key=device_key)
     logging.info("TLS handshake failed as the client certificate is invalid")
 
   @winnforum_testcase
   def test_WINNF_FT_S_SCS_19(self):
-    """Invalid certificate following an approved heartbeat request
-       Checks that SAS UUT response with fatal alert message.
+    """Invalid certificate following an approved heartbeat request.
+
+    Checks that SAS UUT response with fatal alert message.
     """
     device_cert_name = "short_lived_client"
-    expiry_timer = 2  # in minutes
+    cert_duration_minutes = 1  # in minutes
 
-    self.create_short_lived_certificate("CBSD", device_cert_name, expiry_timer)
+    # Create a short lived certificate signed by intermediate CBSD CA.
+    self.createShortLivedCertificate("CBSD", device_cert_name, cert_duration_minutes)
 
+    # Get the absolute path of the short lived certificate created.
     device_cert = self.getCertFilename(device_cert_name + ".cert")
     device_key = self.getCertFilename(device_cert_name + ".key")
-    wait_timer = (expiry_timer * 60) + 10  # In seconds
 
-    # Successful Tls Handshake
+    # Successful TLS Handshake.
     self.assertTlsHandshakeSucceed(self._sas_admin._base_url, ['AES128-GCM-SHA256'],
                                    device_cert, device_key)
     logging.info("TLS Handshake Succeeded")
 
-    # Load the device_a file
-    device_a = json.load(
-      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    # Load the device_a file.
+    device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
 
-    # Load the grant_0 file
-    grant_0 = json.load(
-      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Load the grant_0 file.
+    grant_0 = json.load(open(os.path.join('testcases', 'testdata', 'grant_0.json')))
 
-    with security_testcase.CiphersOverload(self._sas, ['AES128-GCM-SHA256'],
-                                           device_cert,
-                                           device_key):
+    # Register device and grant device with certs(short lived certificates) to SAS UUT.
+    # Ensure the registration and grant requests are successful.
+    with security_testcase.CiphersOverload(self._sas, ['AES128-GCM-SHA256'], device_cert, device_key):
       cbsd_ids, grant_ids = self.assertRegisteredAndGranted([device_a], [grant_0])
+    operation_states = ['GRANTED']
 
-    operation_states = [('GRANTED')]
-
-    with security_testcase.CiphersOverload(self._sas, ['AES128-GCM-SHA256'],
-                                           device_cert,
-                                           device_key):
-      transmit_expire_time = self.assertHeartbeatsSuccessful(cbsd_ids,grant_ids,operation_states)
+    # Send the Heartbeat request for the Grant of CBSD to SAS UUT.
+    with security_testcase.CiphersOverload(self._sas, ['AES128-GCM-SHA256'], device_cert, device_key):
```

Comment:
You could combine this with the previous `with` statement, no?

Reply:
Yes. We   can combine them.           The two statements are combined under the    same "with" block.